pandapower/create.py

```diff
@@ -3886,6 +3886,68 @@ def create_pwl_cost(net, element, et, points, power_type="p", index=None, check=
     return index
 
 
+def create_pwl_costs(net, elements, et, points, power_type="p", index=None, check=True, **kwargs):
+    """
+    Creates an entry for piecewise linear costs for an element. The currently supported elements are
+     - Generator
+     - External Grid
+     - Static Generator
+     - Load
+     - Dcline
+     - Storage
+
+    INPUT:
+        **elements** (iterable of integers) - IDs of the elements in the respective element table
+
+        **et** (string or iterable) - element type, one of "gen", "sgen", "ext_grid", "load",
+                                "dcline", "storage"]
+
+        **points** - (list of list of list) with [[p1, p2, c1], [p2, p3, c2], ...] for each element
+        where c(n) defines the costs between p(n) and p(n+1)
+
+    OPTIONAL:
+        **power_type** - (string or iterable) - Type of cost ["p", "q"] are allowed for active or
+        reactive power
+
+        **index** (int, index) - Force a specified ID if it is available. If None, the index one \
+            higher than the highest already existing index is selected.
+
+        **check** (bool, True) - raises UserWarning if costs already exist to this element.
+
+    OUTPUT:
+        **index** (int) - The unique ID of created cost entry
+
+    EXAMPLE:
+        The cost function is given by the x-values p1 and p2 with the slope m between those points.\
+        The constant part b of a linear function y = m*x + b can be neglected for OPF purposes. \
+        The intervals have to be continuous (the starting point of an interval has to be equal to \
+        the end point of the previous interval).
+
+        To create a gen with costs of 1€/MW between 0 and 20 MW and 2€/MW between 20 and 30:
+
+        create_pwl_cost(net, 0, "gen", [[0, 20, 1], [20, 30, 2]])
+    """
+    if not hasattr(elements, "__iter__") and not isinstance(elements, str):
+        raise ValueError(f"An iterable is expected for elements, not {elements}.")
+    if not hasattr(points):
+        if not len(points) == len(elements):
+            raise ValueError(f"It should be the same, but len(elements) is {len(elements)} "
+                             f"whereas len(points) is{len(points)}.")
+        if not hasattr(points[0]) or len(points[0]) == 0 or not hasattr(points[0][0]):
```

Comment:
What is the purpose of this line? Looks like the second argument of hasattr is missing